### PR TITLE
Update ouroboros-network dependency

### PR DESCRIPTION
**Pull request by newhoggy**
Depends on https://github.com/input-output-hk/cardano-node/pull/4191

### CODE REVIEW

**Comment by coot** [comment]
Are you aware of #4230?

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 1]
```diff
@@ -1770,7 +1769,7 @@ instance (Show versionNumber, ToJSON versionNumber, ToJSON agreedOptions)
       [ "kind" .= String "HandshakeServerError"
       , "reason" .= toJSON err
       ]
-  toObject _verb (TrError e err cerr) =
+  toObject _verb (TrConnectionHandlerError e err cerr) =
     mconcat
       [ "kind" .= String "Error"
```
Comment (mgmeier) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1774]: In PR #4230 this was serialized as `"kind" .= String "ConnectionHandlerError"`. Did you decide to keep the previous serialization / is that change obsolete?

Comment (newhoggy) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs]: Thanks for the pointing this out.  Fixed now.

**Review by deepfire — APPROVED** [review]
Thank you @newhoggy !

**Review by coot — CHANGES_REQUESTED** [review]
The PR #4191 **must** be merged first.  The changes that you're including from `ouroboros-network` enable `NodeToNodeV_10` (p2p). #4191 will be a basis for the initial p2p release, which we want to include as little changes from other components as possible.

File: nix/workbench/shell.nix [hunk 2]
```diff
@@ -88,7 +88,6 @@ in project.shellFor {
   nativeBuildInputs = with pkgs; with haskellPackages; with cardanoNodePackages; [
     cardano-ping
     cabalWrapped
-    db-analyser
```
Comment (deepfire) [inline nix/workbench/shell.nix:91]: @newhoggy, can we avoid removing `db-analyser` -- since that breaks the workbench?

**Review by deepfire — CHANGES_REQUESTED** [review]

**Comment by coot** [comment]
Closing in favour of #4392.

**Pull request closed without merging**